src/core/thread/mle.cpp

```diff
@@ -201,6 +201,14 @@ otError Mle::Start(bool aAnnounceAttach)
     VerifyOrExit(!Get<Radio>().GetPromiscuous(), error = OT_ERROR_INVALID_STATE);
     VerifyOrExit(Get<ThreadNetif>().IsUp(), error = OT_ERROR_INVALID_STATE);
 
+    if (Get<KeyManager>().GetMasterKey() == MasterKey())
```

Comment:
How about generating a random key in the constructor of `KeyManager` instead of the checking and setting here?

Reply:
Could there be a chance that when booting up the system doesn't have enough entropy to generate a truly random master key?

Reply:
I  think we generate a random ext address already in `Mac` constructor which requires crypto rng to be ready.

Reply:
Updated.